src/mbgl/tile/vector_tile.cpp

```diff
@@ -59,6 +59,10 @@ class VectorTileData : public GeometryTileData {
 public:
     VectorTileData(std::shared_ptr<const std::string> data);
 
+    std::unique_ptr<GeometryTileData> clone() const override {
+        return std::make_unique<VectorTileData>(*this);
+    }
+
```

Comment:
Why are we making a copy here? `*TileData` objects should be designed in a read-only, all-const manner (all of its methods are `const`), so we should be able to use shared pointers. Given that they don't hold any resources, except for a `shared_ptr<const std::string>`, they can be destructed in any thread.